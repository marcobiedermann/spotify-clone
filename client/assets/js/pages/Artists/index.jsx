import PropTypes from 'prop-types';
import React from 'react';
import {
  Route,
  Switch,
} from 'react-router-dom';
import ArtistPageContainer from '../../containers/ArtistPage';

const ArtistsPage = ({ match }) => (
  <Switch>
    <Route
      path={`${match.url}/:id`}
      component={ArtistPageContainer}
    />
    <Route
      path={match.url}
      component={() => (
        <div>
          Artists
        </div>
      )}
    />
  </Switch>
);

ArtistsPage.propTypes = {
  match: PropTypes.shape({
    url: PropTypes.string,
  }),
};

ArtistsPage.defaultProps = {
  match: {
    url: '',
  },
};

export default ArtistsPage;
