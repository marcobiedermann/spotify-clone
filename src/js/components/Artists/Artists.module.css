.artists {
  column-gap: 1em;
  display: grid;
  grid-auto-rows: 0;
  grid-template-columns: repeat(auto-fill, minmax(12em, 1fr));
  grid-template-rows: 1fr;
  list-style: none;
  margin: 0;
  overflow-y: hidden;
  padding: 0;
}
