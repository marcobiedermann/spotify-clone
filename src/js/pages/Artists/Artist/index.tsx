import React from 'react';
import { Helmet } from 'react-helmet';
import { Link, RouteChildrenProps, useLocation, useParams } from 'react-router-dom';
import useSWR from 'swr';
import Artist from '../../../components/Artist';
import Error from '../../../components/Error';
import Loader from '../../../components/Loader';

interface Params {
  artistId: string;
}

interface ExternalUrls {
  spotify: string;
}

interface Followers {
  href: null;
  total: number;
}

interface Image {
  height: number;
  url: string;
  width: number;
}

export interface ArtistData {
  external_urls: ExternalUrls;
  followers: Followers;
  genres: string[];
  href: string;
  id: string;
  images: Image[];
  name: string;
  popularity: number;
  type: string;
  uri: string;
}

export type ArtistPageProps = RouteChildrenProps;

function ArtistPage(): JSX.Element {
  const { pathname } = useLocation();
  const { artistId } = useParams<Params>();
  const { data, error } = useSWR<ArtistData>(`/v1/artists/${artistId}`);

  if (error) {
    return <Error>{error.message}</Error>;
  }

  if (!data) {
    return <Loader />;
  }

  const { name } = data;

  return (
    <>
      <Helmet>
        <title>{name}</title>
      </Helmet>
      <Artist {...data} />
      <ul>
        <li>
          <Link to={`${pathname}/albums`}>Albums</Link>
        </li>
        <li>
          <Link to={`${pathname}/related-artists`}>Related Artists</Link>
        </li>
        <li>
          <Link to={`${pathname}/top-tracks`}>Top Tracks</Link>
        </li>
      </ul>
    </>
  );
}

export default ArtistPage;
