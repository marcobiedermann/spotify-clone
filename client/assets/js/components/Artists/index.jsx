import PropTypes from 'prop-types';
import React, { PureComponent } from 'react';
import styles from './style.css';
import Artist from '../Artist';

class Artists extends PureComponent {
  static propTypes = {
    artists: PropTypes.arrayOf(PropTypes.shape),
  };

  static defaultProps = {
    artists: [],
  };

  render() {
    const { artists } = this.props;

    return (
      <ul className={styles.artists}>
        {artists.map(artist => (
          <li key={artist.id}>
            <Artist {...artist} />
          </li>
        ))}
      </ul>
    );
  }
}

export default Artists;
