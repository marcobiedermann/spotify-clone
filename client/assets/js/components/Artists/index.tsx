import classNames from 'classnames';
import React, { FC } from 'react';
import Artist, { ArtistProps } from '../Artist';
import styles from './style.module.css';

export interface ArtistsProps {
  artists: ArtistProps[];
  className?: string;
}

const Artists: FC<ArtistsProps> = (props) => {
  const { artists, className } = props;

  return (
    <ul className={classNames(className, styles.artists)}>
      {artists.map((artist) => (
        <li key={artist.id}>
          <Artist {...artist} />
        </li>
      ))}
    </ul>
  );
};

export default Artists;
