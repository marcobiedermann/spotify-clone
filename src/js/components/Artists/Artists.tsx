import clsx from 'clsx';
import React from 'react';
import Artist, { ArtistProps } from '../Artist';
import styles from './Artists.module.css';

export interface ArtistsProps {
  artists?: ArtistProps[];
  className?: string;
}

function Artists(props: ArtistsProps): JSX.Element {
  const { artists = [], className } = props;

  return (
    <ul className={clsx(className, styles.artists)}>
      {artists.map((artist) => (
        <li key={artist.id}>
          <Artist {...artist} />
        </li>
      ))}
    </ul>
  );
}

export default Artists;
