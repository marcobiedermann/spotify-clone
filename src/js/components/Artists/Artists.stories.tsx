import type { Meta, StoryObj } from '@storybook/react';
import Artists from '.';
import artistFixtures from '../Artist/__fixtures__';

const meta: Meta<typeof Artists> = {
  component: Artists,
  args: {
    artists: [artistFixtures],
  },
};

type Story = StoryObj<typeof Artists>;

const Default: Story = {};

export { Default };
export default meta;
