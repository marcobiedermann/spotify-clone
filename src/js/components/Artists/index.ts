import Artists from './Artists';

export default Artists;
export * from './Artists';
