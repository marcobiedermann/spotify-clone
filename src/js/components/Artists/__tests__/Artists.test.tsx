import { render } from '@testing-library/react';
import React from 'react';
import { expect, it } from 'vitest';
import Artists from '..';

it('renders correctly', () => {
  const { container } = render(<Artists />);

  expect(container.firstChild).toMatchSnapshot();
});
