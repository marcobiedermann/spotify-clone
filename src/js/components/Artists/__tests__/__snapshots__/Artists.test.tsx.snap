// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders correctly 1`] = `
<ul
  class="artists"
>
  <li>
    <figure>
      <img
        alt="string"
        height="300"
        src="https://i.scdn.co/image/ab67616d00001e02ff9ca10b55ce82ae553c8228"
        width="300"
      />
      <figcaption>
        <h2>
          string
        </h2>
      </figcaption>
    </figure>
  </li>
</ul>
`;
